app/api/wallet/mnemonic.ts

```diff
@@ -1,33 +1,44 @@
-import { Network } from '@defichain/jellyfish-network'
-import { MnemonicHdNodeProvider, mnemonicToSeed } from '@defichain/jellyfish-wallet-mnemonic'
+import { Bip32Options, MnemonicHdNodeProvider } from '@defichain/jellyfish-wallet-mnemonic'
+import { EnvironmentNetwork } from '../../environment'
+import { getJellyfishNetwork } from './network'
 import { WalletData, WalletType } from './persistence'
 
-function createProvider (data: WalletData, options: Network): MnemonicHdNodeProvider {
-  const seed = Buffer.from(data.raw, 'hex')
-
-  return MnemonicHdNodeProvider.fromSeed(seed, {
+function getBip32Option (envNetwork: EnvironmentNetwork): Bip32Options {
+  const network = getJellyfishNetwork(envNetwork)
+  return {
     bip32: {
-      public: options.bip32.publicPrefix,
-      private: options.bip32.privatePrefix
+      public: network.bip32.publicPrefix,
+      private: network.bip32.privatePrefix
     },
-    wif: options.wifPrefix
-  })
+    wif: network.wifPrefix
+  }
+}
+
+function createProvider (data: WalletData, network: EnvironmentNetwork): MnemonicHdNodeProvider {
+  if (data.type !== WalletType.MNEMONIC_UNPROTECTED || data.version !== 'v1') {
+    throw new Error('Unexpected WalletData')
+  }
+  const [privKey, chainCode] = JSON.parse(data.raw)
+  return MnemonicHdNodeProvider.fromData({
+    words: [],
+    privKey,
+    chainCode
+  }, getBip32Option(network))
 }
 
-export function createWalletData (mnemonic: string[]): WalletData {
-  const seed = mnemonicToSeed(mnemonic)
-  const hex = seed.toString('hex')
+export function createWalletData (mnemonic: string[], network: EnvironmentNetwork): WalletData {
```

Comment:
Change the arg to EnvironmentNetwork for wallet context API, ease caller syntax and should make sense as Wallet always wrapped inside NetworkContext

Reply:
I should probably refactor this into a `Context`